File: Baekjoon/1912_연속합/B_1912.js
const fs = require('fs')
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')

const N = Number(input[0])
const numbers = input[1].split(' ').map(v => Number(v))

let dp = new Array(N).fill(0)
dp[0] = numbers[0]

let maxNum = dp[0]
for (let i = 1; i < N; i++) {
    dp[i] = Math.max(dp[i - 1] + numbers[i], numbers[i])

    maxNum = Math.max(maxNum, dp[i])
}
console.log(maxNum)

File: baekjoon/1912_연속합/B_1912.py
n = int(input())
numbers = list(map(int, input().split()))

max_num = -1001
dp = [0] * n
dp[0] = numbers[0]
if dp[0] > max_num:
    max_num = dp[0]

for i in range(1, len(numbers)):
    dp[i] = max(dp[i - 1] + numbers[i], numbers[i])
    if dp[i] > max_num:
        max_num = dp[i]

print(max_num)
